docs/guide/coordinates/index.rst

```diff
@@ -0,0 +1,324 @@
+        (111.40782056, 57.1660434, 1.51859559e+11)>
+
+Observer Location Information
+=============================
+
+The `~sunpy.coordinates.frames.Helioprojective`, `~sunpy.coordinates.frames.Heliocentric`
+and `~sunpy.coordinates.frames.HeliographicCarrington` frames are defined by the location of
+the observer. For example in `~sunpy.coordinates.frames.Helioprojective` the
+observer is at the origin of the coordinate system. This information is encoded
+in such observer-based frames as the ``observer`` attribute, which is itself an instance of the
+`~sunpy.coordinates.frames.HeliographicStonyhurst` frame.
+The ``observer`` can be a string for a solar-system body (e.g., "earth" or
+"mars"), and
+`~sunpy.coordinates.ephemeris.get_body_heliographic_stonyhurst` will be used
+with the specified ``obstime`` to fully specify the observer location.  If
+the observer location is not fully specified, or not present at all, most
+transformations cannot be performed.
+The location of the observer is automatically populated from meta data when
+coordinate frames are created using map.
+
+In the case of `~sunpy.coordinates.frames.HeliographicCarrington`, one can specify ``observer='self'`` to indicate that the coordinate itself should be used as the observer for defining the coordinate frame.
+
+It is possible to convert from a `~sunpy.coordinates.frames.Helioprojective`
+frame with one observer location to another
+`~sunpy.coordinates.frames.Helioprojective` frame with a different observer
+location, by converting through `~sunpy.coordinates.frames.HeliographicStonyhurst`, this
+does involve making an assumption of the radius of the Sun to calculate the
+position on the solar sphere. The conversion can be performed as follows::
+
+  # Input coordinate
+  >>> hpc1 = SkyCoord(0*u.arcsec, 0*u.arcsec, observer="earth", obstime="2017-07-26", frame=frames.Helioprojective)
+
+  Define a new Helioprojective frame with a different observer.
+  >>> import sunpy.coordinates
+  >>> hpc_out = sunpy.coordinates.Helioprojective(observer="venus", obstime="2017-07-26")
+
+  Perform the transformation from one to the other.
+  >>> hpc2 = hpc1.transform_to(hpc_out)
+
+An example with two maps, named ``aia`` and ``stereo``::
+
+  >>> hpc1 = SkyCoord(0*u.arcsec, 0*u.arcsec, frame=aia.coordinate_frame)  # doctest: +SKIP
+  >>> hpc2 = hpc1.transform_to(stereo.coordinate_frame)  # doctest: +SKIP
+
+
+Design of the Coordinates Sub-Package
+=====================================
+
+This sub-package works by defining a collection of ``Frames``
+(`sunpy.coordinates.frames`), which exists on a transformation graph, where the
+transformations between the coordinate frames are then defined and registered
+with the transformation graph (`sunpy.coordinates.transformations`). It is also
```

Comment:
```suggestion
with the transformation graph (`sunpy.coordinates`). It is also
```
The transformation graph isn't shown in `sunpy.coordinates.transformations`, which will eventually be turned into a private module anyway.